tests/integration_tests/test_model_training_options.py

```diff
@@ -139,6 +139,46 @@ def test_model_progress_save(skip_save_progress, skip_save_model, tmp_path):
         assert len(training_checkpoints) > 0
 
 
+@pytest.mark.parametrize("save_ludwig_config_with_model_weights", [False, True])
+def test_ludwig_config_save(save_ludwig_config_with_model_weights, tmp_path):
+    input_features, output_features = synthetic_test_data.get_feature_configs()
```

Comment:
@sanjaydasgupta If you do not mind, adding type hints would be helpful for the overall code quality, even if we have not been doing it consistently before (it will set a good pattern for the future).  Thank you!